google-cloud-spanner/src/main/java/com/google/cloud/spanner/Spanner.java

```diff
@@ -35,8 +38,12 @@
 
   /**
    * Closes all the clients associated with this instance and frees up all the resources. This
-   * method does not block. Return future will complete when cleanup is done. TODO(user): Add
-   * logging and tracking of leaked sessions.
+   * method will block till it can clean up all the resources. Specifically, it deletes all the
+   * underlying sessions (which involves rpcs) and closes all the gRPC channels. Once this method
+   * called, this object is no longer usable. It is strongly advised to call this method when you
+   * are done with the {@code Spanner} object, typically when your application shuts down. There is
+   * a hard limit on number of sessions in Cloud Spanner and not calling this method can lead to
+   * unused sessions piling up on the backend.
    */
-  ApiFuture<Void> closeAsync();
+  void close();
```

Comment:
Please consider: 
1) Making Spanner implement `java.io.Closeable` interface
2) Implementing finalize() in SpannerImpl and calling close() there.